junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -141,4 +141,100 @@ private static boolean hasParameters(Method method) {
+			return classReplaceToIndicativeSentence(nestedClass);
+		}
+
+		@Override
+		public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
+			return classReplaceToIndicativeSentence(testClass) + getSentenceSeparator(testClass)
+					+ super.generateDisplayNameForMethod(testClass, testMethod);
+		}
+
+		private String classReplaceToIndicativeSentence(Class<?> testClass) {
+			Class<?> classWithEnclosingParent = testClass.getEnclosingClass();
+			DisplayName classWithDisplayName = testClass.getAnnotation(DisplayName.class);
+			DisplayNameGeneration classWithAnnotation = testClass.getAnnotation(DisplayNameGeneration.class);
+
+			if (classWithEnclosingParent == null) {
+				if (classWithDisplayName != null)
+					return classWithDisplayName.value();
+				else
+					return generateDisplayNameForClass(testClass);
+			}
+			else {
+				if (classWithAnnotation != null && classWithAnnotation.value() == IndicativeSentencesGenerator.class) {
+					if (classWithDisplayName != null)
+						return classWithDisplayName.value();
+					else
+						return generateDisplayNameForClass(testClass);
+				}
+				else {
+					if (classWithDisplayName != null)
+						return classReplaceToIndicativeSentence(classWithEnclosingParent)
+								+ getSentenceSeparator(testClass) + classWithDisplayName.value();
+					else
+						return classReplaceToIndicativeSentence(classWithEnclosingParent)
+								+ getSentenceSeparator(testClass) + super.generateDisplayNameForNestedClass(testClass);
+
+				}
+			}
+		}
+
+		/**
+		 * Gets the separator for {@link IndicativeSentencesSeparator} when extracting the
+		 * annotation from {@code IndicativeSentencesSeparator}, if it doesn't find it,
+		 * then search for the parent classes, if no separator is found use @code{", "} by default.
+		 *
+		 * @param currentClass the Test Class the separator either custom or default
+		 * @return the indicative sentence separator
+		 * {@code Class.getName()}.
+		 */
+		private String getSentenceSeparator(Class<?> currentClass) {
+			IndicativeSentencesSeparator separator = currentClass.getAnnotation(IndicativeSentencesSeparator.class);
+			if (separator != null)
+				return separator.value();
+
+			Class<?> parentClass = currentClass.getEnclosingClass();
+			if (parentClass != null)
+				return getSentenceSeparator(parentClass);
+
+			return ", ";
+		}
+
+		/**
+		 * Generate a string with simply the name of the test name without the
+		 * classes related divided by a $: AnotherParent$ParentName$Class.
+		 *
+		 * @param testName the Test Class from to extract the parameter types from;
+		 * never {@code blank}.
+		 * @return a string without the $ symbol and the root data that comes from
+		 * {@code Class.getName()}.
+		 */
+		private String removeRootCharacter(String testName) {
+			Preconditions.notBlank(testName, "Input Name parameter should not be blank");
+			String[] testNameSplit = testName.split("\\$");
+			int last = testNameSplit.length - 1;
```

Comment:
Now we may have a different problem... what if there's a `$` in the test name? 😁 (it's possible I think?)

Reply:
Oh I did not thought about it, I'll check what happens

Reply:
Yeah, seems that we can do stuff like this kind of naming:
<img width="793" alt="Screen Shot 2019-12-19 at 3 12 32 PM" src="https://user-images.githubusercontent.com/17280833/71216663-b53c2f00-2280-11ea-9283-50b908c362aa.png">

Also, yes it makes troubles when in case happens  with that method, so i decided to remove ir and use the method `class.simpleName() ` to solve this.